File: Haskell/Atividade-5/exercicio-11.hs
{-
Crie uma funcao com assinatura primeiros :: Int -> [t] -> [t], a qual recebe um numero de elementos, uma lista, e retorna uma lista.
Esta funcao deve retornar uma lista com os n primeiros elementos informados no primeiro parametro.
-}

primeiros :: Int -> [t] -> [t]
primeiros _ [] = []
primeiros 1 (a:b) = [a]
primeiros x (a:b) = a : (primeiros (x - 1) b)

main = do
  print (primeiros 3 [1..20])

File: Prolog/Atividade-16/exercicio-11.pl
/*
Crie uma regra primeiros(N,L1,L2), a qual recebe um numero de elementos N, uma lista L1, e retorna uma lista L2. Esta funcao deve retornar uma lista com os N primeiros elementos informados no primeiro parametro.
*/

primeiros(0,_,[]).
primeiros(N,[H|T],[H|L2]) :- N1 is N - 1, primeiros(N1,T,L2). 
